add nginx-local-ip to dev install instructions (#7057)
medic/cht-core: DEVELOPMENT.md

## Testing locally with devices

> Follow the steps below to use an Android device with a development build of your application. This process is relevant when running v3.5.0 or greater of the Core Framework since it relies on service workers, which requires a valid HTTPS certificate. Use `nginx-local-ip`, `ngrok` or `pagekite` to make your developer build accessible from your Android device by giving it a trusted URL.

Reviewer: ```suggestion
Follow the steps below to use an Android device with a development build of your application. This process is relevant when running v3.5.0 or greater of the Core Framework since it relies on service workers, which require a valid HTTPS certificate. Use `nginx-local-ip`, `ngrok` or `pagekite` to make your developer build accessible from your Android device by giving it a trusted URL.
```

Reviewer: I think 'requires' should be singular here because the antecedent is the "service workers" and not the "Core Framework".

1. Start the api. This can be via docker, grunt, debug, horti, etc.
2. Follow the instructions below to start `nginx-local-ip`, `ngrok` or `pagekite`
3. This will output a generated URL which you can enter into our [android app](https://github.com/medic/medic-android) or browser and connect to your local dev environment.

### nginx-local-ip

> [`nginx-local-ip`](https://github.com/medic/nginx-local-ip) is a local proxy that keeps all traffic local.  There is no latency and no throttling.  If you don't need to share your local CHT instance, it is the preferred method to add a valid SSL certificate instead of using `ngrok` or `pagekite`.

Reviewer: ```suggestion
[`nginx-local-ip`](https://github.com/medic/nginx-local-ip) is a local proxy that keeps all traffic local, and runs without latency or throttling. If sharing your local CHT instance is not required, it is the preferred method to add a valid SSL certificate (rather than `ngrok` or `pagekite`).
```

1. Clone the repo: `git clone https://github.com/medic/nginx-local-ip.git`
1. `cd` into the new directory: `cd nginx-local-ip`

> 1. Assuming your IP is `192.168.0.3`, start `nginx-local-ip` o connect to:

Reviewer 2: I'm think there is something wrong with the grammar here, specially the part _o connect to_. Did you wanted to say _**and** connect to_

Author: nice catch! It was a `to` that was needed to match the other 2 sets of instructions ;) Fixed in 36ca644ca

    * An API running via `grunt` or `horti`, execute `APP_URL=http://192.168.0.3:5988 docker-compose up` and then access it at [https://192-168-0-3.my.local-ip.co/](https://192-168-0-3.my.local-ip.co/)

>     * An API running via `docker`, the ports are remapped, so execute `HTTP=8080 HTTPS=8443 APP_URL=https://192.168.0.3 docker-compose up` and then access it at [https://192-168-0-3.my.local-ip.co:8443/](https://192-168-0-3.my.local-ip.co:8443/)

Reviewer 2: _"An API"_ at the beginning of these 2 sentences sounds like we are talking of some API that is not the one from this project, I would say _"The API"_, but just a suggestion from a non English speaker.

Author: Agreed! fixed in b4b0a5d0b

### Remote Proxies

> `ngrok` and `pagekite` are remote proxies that route local traffic between your client and the CHT via a remote SSL terminator.  While easy and handy, it introduces latency and can sometimes be throttled.  

Reviewer: ```suggestion
`ngrok` and `pagekite` are remote proxies that route local traffic between your client and the CHT via a remote SSL terminator.  While easy and handy, they introduce latency and are sometimes throttled.  
```

#### ngrok

> 1. Create an [ngrok account](https://ngrok.com/), download & install the binary and link your computer to your ngrok account.

Reviewer: ```suggestion
1. Create an [ngrok account](https://ngrok.com/), download and install the binary, then link your computer to your ngrok account.
```

1. Start `ngrok` to connect to:
    * An API running via `grunt` or `horti`, execute `./ngrok http 5988`
    * An API running via `docker`, execute `./ngrok http 443`

> 1. Access the app using the https address shown, eg `https://YOUR-NGROK-NAME.ngrok.io`, replacing `YOUR-NGROK-NAME` with what you signed up with.

Reviewer: ```suggestion
1. Access the app using the https address shown (e.g. `https://YOUR-NGROK-NAME.ngrok.io`, replacing `YOUR-NGROK-NAME` with what you signed up with).
```

> **Note:** `ngrok` sometimes fails due to connection throttling which can cause the service worker cache preload to fail.

Reviewer: ```suggestion
**Note:** The service worker cache preload sometimes fails due to connection throttling (thereby causing an `ngrok` failure at startup).```

I feel like this sentence could be improved but, to be honest, I'm not sure how. I threw up a suggestion though.

Author: I'll trust your word-smithing more than mine ;)

#### pagekite

1. Create a [pagekite account](https://pagekite.net/signup/), download and install the python script.

> 1. Start pagekite, being sure to replace `YOUR-PAGEKIT-NAME` with the URL you signed up for, to connect to:

Reviewer: ```suggestion
1. Start pagekite (be sure to replace `YOUR-PAGEKIT-NAME` with the URL you signed up for) to connect to:
```

    * An API running via `grunt` or `horti`, execute `python2 pagekite.py 5988 YOUR-PAGEKIT-NAME.pagekite.me`
    * An API running via `docker`, execute `python2 pagekite.py 443 YOUR-PAGEKIT-NAME.pagekite.me`

> 1. Access the app using the https address shown, eg `https://YOUR-PAGEKIT-NAME.pagekite.me`

Reviewer: ```suggestion
1. Access the app using the https address shown (e.g. `https://YOUR-PAGEKIT-NAME.pagekite.me`).
```
